File: nodejs_backend/models/attendance/attendance_student_by_dayid.model.js
const mongoose = require('mongoose');
const { Schema } = mongoose;
const db = require('../../config/db');

class AttendanceStudentByDayID {
    constructor(
        moduleID,
        roomName,
        lecturerName,
        subjectName,
        attendance,
        day,
        time
    ) {
        this.moduleID = moduleID;
        this.roomName = roomName;
        this.lecturerName = lecturerName;
        this.subjectName = subjectName;
        this.attendance = attendance;
        this.day = day;
        this.time = time;
    }
}

module.exports = AttendanceStudentByDayID;

File: flutter/lib/model/attendanceStudentByDayIDModel.dart
class AttendanceStudentByDayID {
  final String moduleID;
  final String roomName;
  final String lecturerName;
  final String subjectName;
  final List<String> attendance;
  final String day;
  final String time;

  AttendanceStudentByDayID({
    required this.moduleID,
    required this.roomName,
    required this.lecturerName,
    required this.subjectName,
    required this.attendance,
    required this.day,
    required this.time,
  });

  factory AttendanceStudentByDayID.fromMap(Map<String, dynamic> json) {
    return AttendanceStudentByDayID(
      moduleID: json['moduleID'],
      roomName: json['roomName'],
      lecturerName: json['lecturerName'],
      subjectName: json['subjectName'],
      attendance:
          List<String>.from(json['attendance']), // Ensuring List<String>
      day: json['day'],
      time: json['time'],
    );
  }

  @override
  String toString() {
    return '${moduleID} - ${roomName} - ${lecturerName} - ${subjectName} - ${attendance} - ${day} - ${time}';
  }
}
